### DISCUSSION TITLE
Based gee and python api,Using Savitzky-Golay to calculate NDVI,but it happened some troubles

### DISCUSSION THREAD

**Discussion by lizeshuo1999** [SE-TINF22B2/G2-Schoolify#69] [Q&A]
the code
#SG滤波改善NDVI
# 定义Savitzky-Golay过滤器的参数
degree = 3
kernelSize = 3

#定义时间过滤器，最大时间差为48天
time_Filter = ee.Filter.maxDifference(max_Diff=1000*3600*24*48, left_Field='system:time_start', right_Field='system:time_start')

# 创建Savitzky-Golay过滤函数
def savitzky_golay(image):
    inImage = image.select(['NDVI'])  # 使用 'NDVI' 替换 'NDVI' 以匹配您的数据
    smoothed = inImage.addBands(inImage).convolve(ee.Kernel.square(kernelSize))
    weights = ee.List([
        ee.List([1, 2, 1]),
        ee.List([2, 4, 2]),
        ee.List([1, 2, 1])
    ])
    smoothed = smoothed.updateMask(smoothed.select(0).neq(0))
    outImage = smoothed.divide(ee.Image.constant(16)).multiply(ee.Image(weights))
    return outImage.rename('d_0_NDVI')

# 应用Savitzky-Golay过滤器到NDVI图像集合
SG_NDVI = NDVI.filter(time_Filter).map(savitzky_golay)
map.addLayer(SG_NDVI , 'vis1' , 'SG_NDVI')
problem:'ApiFunction' object has no attribute 'name'
it happened at the 'timeFilter'

**Comment by Schmarvinius** [comment]
Lgtm

**Comment by Kronprinz03** [comment]
I believe in you, you can do it
